libraries/chain/block_log.cpp

```diff
@@ -603,6 +619,43 @@ namespace eosio { namespace chain {
             return pos;
          }
 
+         block_pos_size_t get_block_position_and_size(uint32_t block_num) {
+            uint64_t pos = get_block_pos(block_num);
+
+            if (pos == block_log::npos) {
+               return block_pos_size_t {.position = block_log::npos, .size = 0};
+            }
+
+            assert(head);
+            uint32_t last_block_num = block_header::num_from_id(head->id);
+            assert(block_num <= last_block_num);
```

Comment:
I was debating about this. We have already found the block position for `block_num` so it should be in "good" shape. But as `block_num` comes from Net Plugin and to be paranoid, use EOS_ASSERT just in case.